react-basic-practice/src/pages/ProductDetail/index.jsx

```diff
@@ -59,17 +59,25 @@ const ProductDetail = () => {
 
   const handleAddCart = async () => {
     try {
-      const cartUser = {
-        id: userData.userId,
-        listProducts: [
-          ...listItem.listProducts,
-          {
-            idProduct: product.id,
-            quantity: quantityProduct,
-            name: product.name,
-            price: product.price,
-          }],
-      };
+      let cartUser = {};
+      const c = listItem?.listProducts?.findIndex((item) => item.idProduct === Number(productId));
```

Comment:
Do not name variables like this: a, b, c.... Variable must have a clear meaning.